File: app/src/main/java/org/techtown/moback/activity/SignUpActivity.java
package org.techtown.moback.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.method.PasswordTransformationMethod;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;

import org.techtown.moback.R;

public class SignUpActivity extends AppCompatActivity {

    private ImageButton backPress_btn;
    private Button signUp_btn;
    private ImageButton traililng_btn;

    private EditText id_edit;
    private EditText pw_edit;
    private EditText email_edit;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_up);
        id_edit = findViewById(R.id.id_edit_signup);
        pw_edit = findViewById(R.id.pw_edit_signup);
        email_edit = findViewById(R.id.email_edit_signup);

        backPress_btn = findViewById(R.id.backpress_signup);
        backPress_btn.setOnClickListener(view -> {
            finish();
        });

        signUp_btn = findViewById(R.id.signup_btn_signup);
        signUp_btn.setOnClickListener(view -> {

            //TODO : 회원가입과정
            finish();
        });

        traililng_btn = findViewById(R.id.trailling_btn_signup);


        setTrailingMode();
    }

    //비밀번호 숨김, 해제
    private void setTrailingMode()
    {
        traililng_btn.setOnClickListener(view -> {

            //비공개 -> 공개
            if(pw_edit.getInputType() == 129)
            {
                int position = pw_edit.getSelectionEnd();
                pw_edit.setInputType(131073);
                pw_edit.setSelection(position);
                traililng_btn.setImageDrawable(getDrawable(R.drawable.trailing_icon));
            }
            //공개 -> 비공개
            else
            {
                int position = pw_edit.getSelectionEnd();
                pw_edit.setInputType(129);
                pw_edit.setTransformationMethod(PasswordTransformationMethod.getInstance());
                pw_edit.setSelection(position);
                traililng_btn.setImageDrawable(getDrawable(R.drawable.untrailing_icon));
            }
        });
    }
}

File: app/src/main/java/org/techtown/moback/activity/SignUpActivity.kt
package org.techtown.moback.activity

import android.os.Bundle
import android.text.method.PasswordTransformationMethod
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_sign_in.*
import kotlinx.android.synthetic.main.activity_sign_up.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.async
import kotlinx.coroutines.launch
import org.techtown.moback.R
import org.techtown.moback.server.ServerLibrary

class SignUpActivity : AppCompatActivity() {

    private val TAG = "SignUpActivity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sign_up)

        backpress_signup.setOnClickListener({ view: View? -> finish() })
        signup_btn_signup.setOnClickListener({ view: View? ->

            //회원가입과정
            CoroutineScope(Dispatchers.Default).launch {

                var result = ServerLibrary.registerUser(id_edit_signup.text.toString(), "Test","test", pw_edit_signup.text.toString())

                async(Dispatchers.Main) {
                    if(result)
                    {
                        Toast.makeText(applicationContext, "회원가입 성공", Toast.LENGTH_SHORT).show()
                        finish()
                    }
                    else
                        Toast.makeText(applicationContext, "회원가입 실패", Toast.LENGTH_SHORT).show()
                }
            }
        })
        setTrailingMode()
    }

    //비밀번호 숨김, 해제
    private fun setTrailingMode() {
        trailing_btn_signup.setOnClickListener { view: View? ->

            //비공개 -> 공개
            if (pw_edit_signup!!.inputType == 129) {
                val position = pw_edit_signup.selectionEnd
                pw_edit_signup.inputType = 131073
                pw_edit_signup.setSelection(position)
                trailing_btn_signup.setImageDrawable(getDrawable(R.drawable.trailing_icon))
            } else {
                val position = pw_edit_signup.selectionEnd
                pw_edit_signup.inputType = 129
                pw_edit_signup.transformationMethod = PasswordTransformationMethod.getInstance()
                pw_edit_signup.setSelection(position)
                trailing_btn_signup.setImageDrawable(getDrawable(R.drawable.untrailing_icon))
            }
        }
    }
}